Improve README with specs, links, and contribution guide

Tahoora-Tabassum opened the pull request:
**This pull request enhances the README.md by:**

-  Adding key hardware specification of BeagleBone Black
-  Including official resource links and documentation
-  Providing contribution steps and GSoC mention
-  Highlighting certification information

Tahoora-Tabassum committed:
Improve README with specs, links, and contribution guide

Added BeagleBone Black key specs, useful resources, contribution steps and certification info for clarity.

jadonk commented on README.md:
```diff
@@ -2,4 +2,49 @@ BeagleBone Black
+
+## Getting Started
+To start using BeagleBone Black:
+- [Official Getting started Guide](https://beaglebone.org/getting-started)
```
It is beagleboard.org

jadonk commented on README.md:
```diff
@@ -2,4 +2,49 @@ BeagleBone Black
+## Getting Started
+To start using BeagleBone Black:
+- [Official Getting started Guide](https://beaglebone.org/getting-started)
+- [System Reference Manual (PDF)](./BBB_SRM.pdf)
```
We should have the SRM up on https://docs.beagle.cc now.

jadonk commented on README.md:
```diff
@@ -2,4 +2,49 @@ BeagleBone Black
+To start using BeagleBone Black:
+- [Official Getting started Guide](https://beaglebone.org/getting-started)
+- [System Reference Manual (PDF)](./BBB_SRM.pdf)
+- [Flashing a new OS](https://beaglebone.org/getting-started#update)
```
Again, not beaglebone.org, but beagleboard.org. We redirect the domain, but BeagleBone is a board, BeagleBoard.org is the organization.

jadonk commented on README.md:
```diff
@@ -2,4 +2,49 @@ BeagleBone Black
+
+## Resources
+-[Official Site](https://beagleboard.org)
+-[Hardware Wiki](https://elinux.org/Beagleboard:BeagleBoneBlack)
```
I think elinux.org is probably excessively out-of-date, even by our standards.

Tahoora-Tabassum committed:
Refine README formatting and update links

Corrected BeagleBoard.org links, updated documentation references, and cleaned up README formatting.
